msprime/simulations.py

```diff
@@ -819,57 +825,77 @@ class RecombinationMap(object):
+        recombination occurs only at integer values.
 
     :param list positions: The positions (in bases) denoting the
         distinct intervals where recombination rates change. These can
         be floating point values.
     :param list rates: The list of rates corresponding to the supplied
         ``positions``. Recombination rates are specified per base,
         per generation.
-    :param int num_loci: The maximum number of non-recombining loci
+    :param int num_loci: **This parameter is deprecated**.
+        The maximum number of non-recombining loci
         in the underlying simulation. By default this is set to
         the largest possible value, allowing the maximum resolution
         in the recombination process. However, for a finite sites
         model this can be set to smaller values.
+    :param bool discrete: Whether recombination can occur only at integer
+        positions. When ``False``, recombination sites can take continuous
+        values. To simulate a fixed number of loci, set this parameter to
+        ``True`` and scale ``positions`` to span the desired number of loci.
     """
-    DEFAULT_NUM_LOCI = 2**32 - 1
-    """
-    The default number of non-recombining loci in a RecombinationMap.
-    """
-    def __init__(self, positions, rates, num_loci=None):
-        if num_loci is None:
-            num_loci = self.DEFAULT_NUM_LOCI
+    def __init__(self, positions, rates, num_loci=None, discrete=False):
+        if num_loci is not None:
+            if num_loci == positions[-1]:
+                warnings.warn(
+                        "num_loci is deprecated and should not be used. "
+                        "The default behavior is that the number of loci "
+                        "is equal to the length of the sequence.",
+                        DeprecationWarning)
+            else:
+                raise ValueError(
+                        "num_loci does not match sequence length. "
+                        "To set a discrete number of recombination sites, "
+                        "scale positions to span the desired number of loci "
+                        "and set discrete=True")
         self._ll_recombination_map = _msprime.RecombinationMap(
-            num_loci, positions, rates)
+            positions, rates, discrete)
 
     @classmethod
-    def uniform_map(cls, length, rate, num_loci=None):
+    def uniform_map(cls, length, rate, num_loci=None, discrete=False):
         """
         Returns a :class:`.RecombinationMap` instance in which the recombination
         rate is constant over a chromosome of the specified length. The optional
-        ``num_loci`` controls the number of discrete loci in the underlying
-        simulation, and is by default large enough to be effectively be
-        a continuous model.
+        ``discrete`` controls whether recombination sites can occur only on integer
+        positions or can take continuous values. The optional ``num_loci`` is
```

Comment:
It's not so much deprecated as removed. "The legacy ``num_loci`` option is no longer supported and should not be used".